### DISCUSSION TITLE
Where to put KML files for import on iPhone?

### DISCUSSION THREAD

**Discussion by 4tegs** [organicmaps/organicmaps#1261] [Q&A]
When using Maps.me I can easily import KML files by using iTunes -> File Sharing function. 
Trying to do so with Organic Maps doesn't work. 
I tried the the same files on both apps.

Any idea?

**Comment by biodranik** [comment]
Copy `bookmarks` folder to your computer using File Sharing (to get whatever bookmarks you already have), put your additional KML files there, then upload the `bookmarks` folder back to the device.

**Reply by 4tegs** [reply]
Hi Biodranik, thanks a lot for your immediate answer. 

Just as an information: Uploading <bookmarks> works well, downloading isn't - I can download files only. 
For me that isn't an issue, as I can maintain all my bookmarks on my PC.

**Reply by biodranik** [reply]
In Mac OS X, you can drag the bookmarks folder from the app into Finder, and all XML files will be copied. How does it work on PC?

**Reply by 4tegs** [reply]
I'm using Windows 10 Professional, iTunes 12.11.0.26, new iPhone SE iOS 14.7.1

iTunes File Sharing (App) -> PC: Drag & Drop files: Works // Folder: Doesn't work.
PC -> iTunes File Sharing (App): Drag & Drop files: Works // Folders: Works.

**Reply by 4tegs** [reply]
BTW: I think this is the reason why Maps.me uses an import function. I simply drag & drop a KML file to the "root" File Sharing folder of the app. As soon as I open Bookmarks, Maps.me recognizes that there is a KML file and imports it into the bookmarks folder (and finally deleting my original). This makes sure that the bookmarks folder is always well maintained and not destroyed by importing a new bookmarks folder each time one wants to import a new KML.

**Reply by biodranik** [reply]
Does the similar import from the root folder work with Organic Maps? We need to make it work if it's not :)

**Reply by 4tegs** [reply]
Nope

**Reply by 4tegs** [reply]
In case you need me for any type of test: Just let me know!

**Reply by 4tegs** [reply]
I need to do an update. 
While copy & paste do not work, there is a button in iTunes where in "File Sharing - Organic Maps" one can click the item (bookmarks), then scroll down to the bottom of the list (not visible at the top where one marks "bookmarks") then click "save". 
That works!
